pkg/kv/kvclient/kvcoord/BUILD.bazel

```diff
@@ -70,29 +70,36 @@ go_library(
     ],
 )
 
-# keep
-go_library(
-    name = "kvcoord",
-    srcs = [":mocks_transport"],
-    embed = [":kvcoord_base"],
-    visibility = ["//visibility:public"],
-    deps = [
-        "@com_github_golang_mock//gomock",
-        "@org_golang_google_grpc//metadata",
-    ],
-)
-
 gomock(
-    name = "mocks_transport",
+    name = "mock_kvcoord",
     out = "mocks_generated.go",
     interfaces = [
         "Transport",
     ],
-    library = ":kvcoord_base",
+    library = ":kvcoord",
     package = "kvcoord",
     self_package = "github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord",
 )
 
+go_library(
+    name = "with-mocks",
+    srcs = [":mock_kvcoord"],
+    embed = [":kvcoord"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@com_github_golang_mock//gomock",
+    ],
+)
+
+# This noop target is a workaround for https://github.com/bazelbuild/bazel-gazelle/issues/1078.
+#
+# gazelle:resolve go github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord //pkg/kv/kvclient/kvcoord:noop
+go_library(
+    name = "noop",
+    importpath = "noop",
```

Comment:
This is pretty ugly. I think you're having to do this because `pkg/kv/kvclient/kvcoord` is a package that contains both [internal and external tests](https://github.com/bazelbuild/rules_go/blob/master/go/core.rst#internal-test-example), right?

The blessed way to do that would be to have two separate test targets, one for the external tests and one for the internal tests. Unfortunately Gazelle doesn't do that for us. (Not sure why not.) I'm hesitant to do this ugly thing because now we're trying to get `rules_go` to do something it doesn't want to do so our tests don't fail.

Bare minimum, you can add a new `go_library(name = "noop", importpath = "noop", visibility = ["//pkg:__subpackages__"])` in `build/bazelutil/BUILD.bazel` and reference that everywhere we want an empty `go_library`.

Reply:
Done. 

> The blessed way to do that would be to have two separate test targets, one for the external tests and one for the internal tests. Unfortunately Gazelle doesn't do that for us. (Not sure why not.)

Having separate test targets used to be a thing before, and was changed at some point: https://github.com/bazelbuild/bazel-gazelle/issues/102. Unsure why, brevity maybe? I assume `go_test` is internally capturing the fact that there are both internal and external tests specified in the same target, I think what I'm asking in  bazelbuild/bazel-gazelle#1078 then is for gazelle to construct the right dependencies by looking at the import path, rather than just the label. Curious to see what they think of this upstream.